clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/LeastLoadedClientRoutingStrategy.java

```diff
@@ -23,20 +23,20 @@ public LeastLoadedClientRoutingStrategy(InstanceHealthMonitor instanceHealthMoni
   }
 
   @Override
-  public List<String> getReplicas(long requestId, List<String> replicas, int requiredReplicaCount) {
+  public List<String> getReplicas(long ignored, List<String> replicas, int requiredReplicaCount) {
     if (replicas.isEmpty()) {
       return Collections.emptyList();
     }
-    int replicaCnt = replicas.size();
-    int startPos = (int) (requestId % replicaCnt);
     List<String> availReplicas = new ArrayList<>();
-    for (int i = 0; i < replicaCnt; ++i) {
-      String replica = replicas.get((i + startPos) % replicaCnt);
+    /**
+     * For even distribution, we need to shuffle the replicas.
+     */
+    Collections.shuffle(replicas);
```

Comment:
Looking at the code, `requestId` only made a difference if the pending request count is the same for multiple replicas as it immediately does a sort after this. so, when there is not much traffic, if the pending request counter numbers are 0 (or similar due to some reasons), then the shuffle comes to rescue?

Reply:
Each client only has the local view, so if the qps is low, the pending request count for every node can be 0 for most of time as you mentioned.

Request-id-based replica assignment will always assign the same group of replicas when the reminder is same, which results in the consistent request banding in the backend, and reshuffle will guarantee some randomness for every request to achieve even distribution.

Reply:
Makes sense thanks.